src/kirby/components/form-field/input/input.component.ts

```diff
@@ -0,0 +1,32 @@
+import { ChangeDetectionStrategy, Component, Input, HostBinding } from '@angular/core';
+
+@Component({
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  // tslint:disable-next-line:component-selector
+  selector: 'input[kirby-input]',
+  styleUrls: ['./input.component.scss'],
+  template: '',
+})
+export class InputComponent {
+  private static typeToInputmodeMap = {
+    number: 'decimal',
+    search: 'search',
+  };
+  @Input() set type(value: string) {
+    this._inputmode = InputComponent.typeToInputmodeMap[value];
+  }
+  @HostBinding('attr.inputmode')
+  _inputmode: string;
+
+  @HostBinding('class.state-success')
+  _isSuccess: boolean;
```

Comment:
Do we use _ prefix for public fields?

Reply:
I think it is because, it is an internally used public field.